src/Chartjs/assets/src/controller.ts

```diff
@@ -59,6 +59,15 @@ export default class extends Controller {
         this.dispatchEvent('connect', { chart: this.chart });
     }
 
+    disconnect() {
+        if (this.chart) {
+            this.chart.destroy();
+            this.chart = null;
+        }
+
+        this.dispatchEvent('disconnect', { chart: this.chart });
+    }
+
```

Comment:
```suggestion
    disconnect() {
        this.dispatchEvent('disconnect', { chart: this.chart });
        
        if (this.chart) {
            this.chart.destroy();
            this.chart = null;
        }
    }
```

In that order, you let the possibility for listeners to interact with this.chart _before_ destroying the instance.